### PR TITLE
Add new firefox xpi extensions support

### PR DESCRIPTION
**Pull request by alexander-schranz**
This is a rebased version of #596. Thx goes here to @maxvgi which did work on this.

### CODE REVIEW

File: lib/Firefox/FirefoxProfile.php [hunk 1]
```diff
@@ -185,44 +185,65 @@ public function encode()
...
+            ($second_marker_pos_in_hex_string + strlen($object_identifier_hex_marker)) / 2;
+
+        $common_name_string_length = ord($mozilla_rsa_binary_data[$common_name_string_position_in_binary + 1]);
+        $addon_common_name = substr(
+            $mozilla_rsa_binary_data,
+            $common_name_string_position_in_binary + 2,
+            $common_name_string_length
+        );
+
+        if (!preg_match('/^\\{[0-9a-f-]{36}\\}$/', $addon_common_name)) {
```
Comment (alexander-schranz) [inline lib/Firefox/FirefoxProfile.php:226]: This does currently not work I'm getting `uBlock0@raymondhill.net` but the regex does not match.

Comment (alexander-schranz) [reply lib/Firefox/FirefoxProfile.php]: I removed this check as it didn't work and not sure what I should check here.

**Comment by alexander-schranz** [comment]
Ready for review. Let me know what I can do else to get this merged.

**Review by williamdes — APPROVED** [review]

**Comment by OndraM** [comment]
I further refactored the extension installation and re-signed the xpi extension. Once the CI finishes we can finally merge this :tada: . Thanks for patience!

**Comment by alexander-schranz** [comment]
Nice. Thank you 👍

**Review by OndraM — APPROVED** [review]

**Pull request merged**